### ISSUE TITLE
The vcd_nsxt_app_port_profile resource type is unusable

### ISSUE THREAD

**Issue by GeorgiosGoniotakis** [vmware/terraform-provider-vcd#802]
Hello there,

We are using the `vcd_nsxt_app_port_profile` resource to create Application Port Profiles. Then, we export the URNs of these application port profiles and use them in our firewall configuration by utilizing the `app_port_profile_ids` property of the `vcd_nsxt_firewall` resource. 

It seems that NSX is performing some kind of update to the URNs of the application port profiles overnight which causes their replacement and the replacement of any of the resources which are referencing them. I am opening this ticket to save potentially dozens of hours of frustration to anyone who is currently using or planning to use this resource.

As it is now, creating or using the URN of an Application Port Profile (for example by using its data source) in any other resource is not possible.

### Affected Resource(s)
- vcd_nsxt_app_port_profile resource and data source
- Any other resource which references the application port profile IDs

### Terraform Configuration Files
Any standard application port profile definition can trigger this issue. Kindly provision an application port profile using the HCL code below and run a `terraform plan` command after a couple of days.

```hcl
resource "vcd_nsxt_app_port_profile" "custom-app" {
  org = "my-org"
  vdc = "my-nsxt-vdc"

  name        = "custom app profile"
  description = "Application port profile for custom application"

  scope = "TENANT"

  app_port {
    protocol = "ICMPv6"
  }

  app_port {
    protocol = "TCP"
    port     = ["2000", "2010-2020", "12345", "65000"]
  }

  app_port {
    protocol = "UDP"
    port     = ["40000-60000"]
  }
}
```

### Expected Behavior
The URNs of the application port profiles should not change without replacing them. No changes should be introduced to the plan without a code change.

### Actual Behavior
The URNs of the application port profiles change overnight. This causes the replacement/update of them and of any resources that are referencing them.

### Steps to Reproduce
1. Use the code snippet above to create an application port profile
2. Wait for 24-48 hours
3. Run `terraform plan` without making any changes to the code or the state.
4. Notice that the produced plan is proposing a resource replacement.

**Comment by claudineimatos** [comment]
I am having the same issue here. Have created a few app_port_profile but I'm also using built-in one.  A few months ago I have noticed that the IDs changed but thinked it was due to some migration of edge gateway. Today I was trying to create a new rule and found that all the URNs changed again, both built-in and custom. The last time it took me a few hours to identify the new IDs and fix the TF, but this is not something I can run over every time I need to make changes on the platform.
A fix is needed ASAP.

**Comment by mkriegs** [comment]
Its something in vcd side with snyc of the NSXT Controller. so its nothing on terraform side. i notice the same think you did.

**Comment by Didainius** [comment]
Yes, indeed. We don't really know how this could be fixed from Terraform side as these shifting IDs cause trouble for us as well.

**Comment by Didainius** [comment]
We cannot fix this bug from `terraform-provider-vcd` as the ID shift problem is in VCD itself. VCD team knows that and a fix should be provided in future. I am closing this issue for now as we believe no additional work should be required for this provider once VCD bug is fixed.

**Comment by Didainius** [comment]
Just an update for those wandering here. This was a VCD problem and it is reported to be fixed in VCD 10.4.0.

**Comment by GeorgiosGoniotakis** [comment]
Thank you for confirming @Didainius 😄

**Comment by ebuh** [comment]
Fixed in 10.3.3.2 as well
